segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -897,11 +898,13 @@ private boolean reconcileMetadata(LogMetadata metadata) {
      * @throws DurableDataLogException  If another kind of exception occurred. See {@link #persistMetadata}.
      */
     @VisibleForTesting
-    void overWriteMetadata(LogMetadata metadata) throws DurableDataLogException {
+    void overWriteMetadata(LogMetadata metadata, boolean overwrite) throws DurableDataLogException {
```

Comment:
We have a `forceMetadataOverWrite()` method in `DebugBookkeeperLogWrapper`. I wonder if we could use that instead of changing the `BookkeeperLog` class.

Reply:
reverted changes to this method.